### ISSUE TITLE
question_numeric with tolerance

### ISSUE THREAD

**Issue by tvedebrink** [rstudio/learnr#530]
The `question_numeric` does not take a tolerance argument, but the comparision of the answer and user input is evaluated using a tolerance of `1e-10`: https://github.com/rstudio/learnr/blob/master/R/question_numeric.R#L99
It would be nice if this tolerance could be specified, e.g. `tol = 1e-10`.
This could be achived by replacing L99 by: `if (isTRUE(all.equal(ans_val, value, tolerance = question$options$tol)))`, adding `tol = tol` after L59 and adding `tol = 1e-10` as the default argument to `question_numeric`

**Comment by gadenbuie** [comment]
Thanks @tvedebrink, this is a great suggestion and it looks like you have a good handle of what is needed to make the change. My only comment on what you've sketched out is that I'd prefer to use the full `tolerance` rather than `tol` for the argument name. Do you want to submit a PR to add this argument to `question_numeric()`?

**Comment by schloerke** [comment]
Could we have the default value be `tolerance = NULL` and when comparing should we use `waldo::compare`?

```r
compare_text <- waldo::compare(ans_val, value, tolerance = question$options$tolerance)
if(length(compare_text) == 0) {
    # answer is the same
}
```
?

**Comment by gadenbuie** [comment]
That seems a bit much given the `question_numeric()` context. Even `all.equal()` is a bit much. I'm inclined to use this instead:

```r
if (diff(range(c(ans_val, value))) <= question$options$tolerance) {
  # answer is the same
}
```

**Comment by gadenbuie** [comment]
This was fixed in #657 where `question_numeric()` gained a `tolerance` argument. The tolerance is applied globally to all answers, but for varying tolerance by `answer()` you may use the new `answer_fn()` answer type.
